src/main/java/uk/gov/ons/ssdc/supporttool/endpoint/DeactivateUacEndpoint.java

```diff
@@ -49,6 +49,12 @@ public DeactivateUacEndpoint(
   public void deactivateUac(
       @PathVariable("qid") String qid,
       @Value("#{request.getAttribute('userEmail')}") String userEmail) {
+
+    authUser.checkUserPermission(
+            userEmail,
+            uacQidLinkOpt.get().getCaze().getCollectionExercise().getSurvey().getId(),
+            DEACTIVATE_UAC);
```

Comment:
I don't think this check will work here, it needs to fetch the uacQidLink first to know what survey it's checking permissions for

Reply:
I spoke to Khalid about that and we decided it might be best to check global permission first to see if the user can actually perform the action, then get the survey/collection exercise needed, then check if the use has permission to perform the action on that specific survey

Reply:
That's not quite how the permissions work at the moment, the global permissions are just for superuser/site wide functions. A user with just permissions on a specific survey won't also have global permissions, they're additive. 

I think in these cases there isn't really an issue with finding the qid/collection-exercise before the permissions check, perhaps just with a comment clarifying why it has to happen first. If the qid/collection-exercise does exist, then the user still gets a permission denied response if they don't have permissions, otherwise if the qid/collection-exercise doesn't exist they get a 404. The permissions check should always be done as soon as possible, but in cases like these we do need to do a look-up first.